alerts/alert.go

```diff
@@ -0,0 +1,71 @@
+package alerts
+
+import "github.com/MohawkTSDB/mohawk/backend"
+
+const (
+	CLOSED_INTERVAL RangeIntervalType = iota
+	LEFT_CLOSED_INTERVAL
+	RIGHT_CLOSED_INTERVAL
+)
+
+const (
+	PENDING AlertState = iota
+	FIRING
+	RESOLVED
+)
+
+// Denotes the range interval type i.e closed interval are intervals of the form [from, to] whereas
+// open intervals are intervals of the form [-inf, to] (RIGHT_CLOSED_INTERVAL) or [from, inf] (LEFT_CLOSED_INTERVAL).
+type RangeIntervalType int
+
+// denotes the state of the active alert
+type AlertState int
+
+type Range struct {
+	from float64
+	to   float64
+	t    RangeIntervalType
+}
+
+type Alert struct {
+	// id is a valid metric label - there can be only one alert per metric.
```

Comment:
that should not be true !!

I want to send on message when cpu_usage > 80% and a new message when cpu_usage > 95% !!

e.g.
```
alert list:
   _system:
        cpu_usage_check_it:
             type: higher then
             limit: 85
             message: blah blha
        cpu_usage_oy_vey:
             metric: cpu_usage
             type: higher then
             limit: 95
             message: blah blha
```

Reply:
p.s.

The list above is just an example, not the exect way I think the list should look like (_system is the tenant name)

Reply:
p.p.s

one day we will also want to support combined alerts ( e.g. if alert X AND alert Y are on the alert Z is on ... ) so think about it when you do think [ currently I think this should be implemented in another loop running immediately after the main alerts loop ... but it's very open in this stage ]